### PR TITLE
[experimental] Add linting step in test pipeline

### PR DESCRIPTION
**Pull request by yamafaktory**
In order to enforce a good code quality, I think we should make use of linting step in the Travis CI pipeline. This PR tries to solve that via a really small set of basic ESLint rules. In the end, this allows us to drop a bunch of unused code.

/cc @vjeux that was what I was thinking of in https://github.com/prettier/prettier/pull/1056#issuecomment-292723656 😄.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  "rules": {
+    "func-call-spacing": 2,
+    "guard-for-in": 2,
+    "no-debugger": 2,
+    "no-dupe-args": 2,
+    "no-dupe-keys": 2,
+    "no-duplicate-case": 2,
+    "no-multi-spaces": 2,
+    "no-self-compare": 2,
+    "no-unused-vars": 2,
```
Comment (vjeux) [inline .eslintrc]: It seems like this is the only rule that actually caught something, no?

**Comment by lydell** [comment]
Why include rules about spaces when we run prettier on the code? :)

File: .eslintrc (hunk 1 above)
Comment (yamafaktory) [inline .eslintrc]: The `no-debugger` also caught something. The `no-multi-spaces` was also involved but as @lydell righteously mentioned it, we shouldn't interfere on the code style itself as this is where Prettier shines actually 👍.

**Comment by yamafaktory** [comment]
@lydell I totally agree on that point 👍. I've just refined it accordingly. This new test step is pretty fast and is here just to ensure not to push dead / unused code.

**Comment by yamafaktory** [comment]
@vjeux I added the `no-console` rule in order to stay consistent with the `no-debugger` one. This way, with the insertion of disable comments where console log/warn/error are necessary (*prettier binary / options / sync-flow-tests*), this will only affect new debug code inserted by mistake.

File: bin/prettier.js [hunk 2]
```diff
@@ -36,13 +36,15 @@ const argv = minimist(process.argv.slice(2), {
   alias: { help: "h", version: "v", "list-different": "l" },
   unknown: param => {
     if (param.startsWith("-")) {
+      // eslint-disable-next-line no-console
```
Comment (vjeux) [inline bin/prettier.js]: If you have to add 23 comments to disable a lint warning, we should instead remove the lint warning. It doesn't add much value and is way more annoying

Comment (lydell) [inline bin/prettier.js]: Another way would be to add `/* eslint-disable no-console */` at the top of bin/prettier.js and scripts/sync-flow-tests.js.

**Comment by vjeux** [comment]
So, the only useful rule here is the no-unused-var. As much as possible, I'd like to avoid adding infrastructure unless it's extremely useful. In this case, it doesn't seem so. I think it's fine to run it once in a while and remove all the unused code.

Could you remove the lint integration from this pull request and send fixes for all the unused variables/debugger? I'll merge it.

Then, if you want for posterity you can create a new PR for the lint integration, but I don't want to merge it.

Thanks for doing this work!

File: bin/prettier.js (hunk 2 above)
Comment (yamafaktory) [inline bin/prettier.js]: Moving the disabling comment on top of the aforementioned files will not prevent any new wrong console insertion 😄 .

**Comment by yamafaktory** [comment]
@vjeux ok, absolutely no problem! I'll not create a PR for *posterity* if you consider it useless in the end for the project. I've cleaned up the current PR to match your request.

**Comment by yamafaktory** [comment]
@vjeux I synced with current master, feel free to merge it or drop it.

**Comment by vjeux** [comment]
I told you I would like to merge all the changes that were found by the lint rule but not the lint rule itself. If you would like to remove the infrastructure for the lint rule, I'll merge it in. Thanks!

**Comment by yamafaktory** [comment]
@vjeux Oh, I'm deeply sorry, I only partially got it (I thought keeping the script in here for further testing was ok...). This should be clean now 👍.

File: yarn.lock [hunk 3]
```diff
@@ -12,9 +12,34 @@ acorn-globals@^3.1.0:
   dependencies:
```
Comment (vjeux) [inline yarn.lock]: Please revert this file.

File: package.json [hunk 4]
```diff
@@ -36,11 +36,11 @@
...
   },
   "scripts": {
-    "test": "jest",
+    "build:docs": "rollup -c docs/rollup.config.js",
     "format": "./bin/prettier.js --write",
-    "format:single": "npm run format -- src/printer.js",
     "format:all": "npm run format -- index.js src/*.js bin/*.js",
-    "build:docs": "rollup -c docs/rollup.config.js"
+    "format:single": "npm run format -- src/printer.js",
+    "test": "jest"
```
Comment (lydell) [inline package.json]: These changes don't seem very related to this PR, do they? :)

Comment (vjeux) [inline package.json]: There was probably a lint rule to sort those.

File: yarn.lock (hunk 3 above)
Comment (yamafaktory) [inline yarn.lock]: Done, sorry I forgot that one.

File: package.json (hunk 4 above)
Comment (yamafaktory) [inline package.json]: No, it was done on purpose, anyway I reverted it.

**Comment by vjeux** [comment]
The lockfile is still there :(

**Comment by yamafaktory** [comment]
Huh, weird... but should be fixed now!

**Comment by vjeux** [comment]
Thanks!

**Pull request merged**